src/rust-crypto/backup.ts

```diff
@@ -106,6 +106,27 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
         };
     }
 
+    /**
+     * Checks if the provided backup info matches the given private key.
+     *
+     * @param info - The backup info to check.
+     * @param backupDecryptionKey - The `BackupDecryptionKey` private key to check against.
+     * @returns `true` if the private key can decrypt the backup, `false` otherwise.
+     */
+    private backupInfoMatchesBackupDecryptionKey(
+        info: KeyBackupInfo,
+        backupDecryptionKey: RustSdkCryptoJs.BackupDecryptionKey,
+    ): boolean {
+        if (info.algorithm !== "m.megolm_backup.v1.curve25519-aes-sha2") {
```

Comment:
shouldn't we also check that `backupDecryptionKey.megolmV1PublicKey.algorithm` is `"m.megolm_backup.v1.curve25519-aes-sha2"`  ?